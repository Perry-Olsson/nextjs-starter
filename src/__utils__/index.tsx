import React, { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "../components/theme";
import { GlobalStyle } from "../pages/_app";

const AllTheProviders: FC = ({ children }) => {
  return (
    <ThemeProvider>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
