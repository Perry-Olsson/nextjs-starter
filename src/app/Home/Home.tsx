import { FC } from "react";
import styled from "styled-components";

export const Home: FC = () => {
  return (
    <Container>
      <Title>Home Page</Title>
    </Container>
  );
};

const Container = styled.div`
  display: flex;
  justify-content: center;
  margin-top: 50px;
`;

const Title = styled.h1`
  color: ${({ theme }) => theme.text.primary};
`;
