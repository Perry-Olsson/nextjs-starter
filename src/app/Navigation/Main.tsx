import styled from "styled-components";
import constants from "components/constants";
import { Link, Button, Flex, useThemeToggler } from "components";

export const MainNav = () => {
  const toggle = useThemeToggler();

  return (
    <Container as="nav">
      <div>
        <Link href="/">Home</Link>
        <Link href="/about">About</Link>
        <Link href="/services">Services</Link>
      </div>
      <div>
        <Button onClick={toggle}>Change Theme</Button>
      </div>
    </Container>
  );
};

const Container = styled(Flex)`
  flex-direction: row;
  justify-content: space-between;
  padding: 0 2rem;
  height: ${constants.mainNav.height};
  border-bottom: solid 1px;
`;
