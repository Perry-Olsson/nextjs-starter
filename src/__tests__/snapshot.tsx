import { createSnap } from "../__utils__/index";
import { MainNav } from "../app/Navigation";
import Home from "../pages/index";

describe("App", () => {
  it("Home Page", () => {
    const tree = createSnap(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Navigation", () => {
    const tree = createSnap(<MainNav />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
