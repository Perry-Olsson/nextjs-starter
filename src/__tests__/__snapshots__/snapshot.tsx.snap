// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Home Page 1`] = `
<main
  className="Main-sc-1h07s3x-0 egPrzL"
>
  <h1>
    Home Page
  </h1>
</main>
`;

exports[`App Navigation 1`] = `
<nav
  className="Flex-sc-14opyce-0 Main__Container-sc-1bu84wq-0 hvJiLq jWAtVQ"
>
  <div>
    <a
      className="Link__AnchorTag-sc-y5hrub-0 TUJiH"
      href="/"
      onClick={[MockFunction]}
    >
      Home
    </a>
    <a
      className="Link__AnchorTag-sc-y5hrub-0 TUJiH"
      href="/about"
      onClick={[MockFunction]}
    >
      About
    </a>
    <a
      className="Link__AnchorTag-sc-y5hrub-0 TUJiH"
      href="/services"
      onClick={[MockFunction]}
    >
      Services
    </a>
  </div>
  <div>
    <button
      className="Button-sc-11v6kpx-0 eacQDx"
      onClick={[Function]}
    >
      Change Theme
    </button>
  </div>
</nav>
`;
